test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -282,6 +282,16 @@ protected T createUnmappedInstance(String name, Map<String, Object> metadata) {
         return createTestInstance(name, metadata);
     }
 
+    /**
+     * The NamedWriteable class for agg tests _should_ be InternalAggregation so this is being set by default,
+     * but implementors can change if necessary
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    protected Class<T> categoryClass() {
```

Comment:
Could this be `final`?

Reply:
It could be.  I left it non-final in case an implementor needs to override it for some reason.  But I suppose with the generic bounds on <InternalAggregation> it's pretty definitive what the class should be...

Reply:
I do think this should be `final` then. I guess the `suppresswarnings` means that the bounds on `AbstractNamedWriteableTestCase` are wrong.

Reply:
Yeah to be honest I really have no idea what's going on with the generic... scratched my head over it for a while and just gave up :/ 

I'll make `final` :+1: